crates/bevy_render/src/render_graph/nodes/pass_node.rs

```diff
@@ -216,13 +217,30 @@ where
                         continue;
                     };
 
-                    // get an ordered list of entities visible to the camera
-                    let visible_entities = if let Some(camera_entity) = active_cameras.get(&camera_info.name) {
-                        world.get::<VisibleEntities>(camera_entity).unwrap()
+                    let camera_entity = if let Some(camera_entity) = active_cameras.get(&camera_info.name) {
+                        camera_entity
                     } else {
                         continue;
                     };
 
+                    // get an ordered list of entities visible to the camera
+                    let visible_entities = world.get::<VisibleEntities>(camera_entity).unwrap();
+
+                    // get camera viewport and apply it
+                    if let Ok(viewport) = world.get::<Viewport>(camera_entity) {
+                        let origin = viewport.physical_origin();
+                        let size = viewport.physical_size();
+                        let (min_depth, max_depth) = viewport.depth_range().into_inner();
+                        render_pass.set_viewport(
+                            origin.x, origin.y,
+                            size.x, size.y,
+                            min_depth, max_depth,
+                        );
+                    } else {
+                        // a camera requires a valid viewport
+                        continue;
```

Comment:
It would be nice to give some indication that a viewport is missing. Might save someone a day of debugging.

Reply:
You're right, and thinking about it, this would be a serious bug. I think a panic (like in the case of VisibleEntities) is justified, at least for now.